### PR TITLE
Allow different verbs payload

### PR DESCRIPTION
**Pull request by nahime0**
This PR introduces support to POST requests. 

Two new options has been introduces:
- method: can be either get or post
- payload: the json encoded data to use for the stress test

The two options will be forwarded to k6 using the already defined options environment variable. 
The run.js used by k6 has been adapted to use get or post based on the method, and to use the payload variable only if the required method is post.

@nunomaduro I haven't written any test. I don't know if you want to use a different test suite, as the default one uses the same stress object for all requests (defined in the beforeEach of Pest.php).

### CODE REVIEW

**Review by devajmeireles — APPROVED** [review]

File: README.md
Comment (devajmeireles) [inline README.md:1]: The official Stressless doc was moved to the Pest doc: https://pestphp.com/docs/stress-testing

File: src/Factory.php [hunk 1]
```diff
@@ -59,6 +71,48 @@ public function duration(int $seconds): self
...
+     */
+    public function get(): self
+    {
+        return $this->method('get');
+    }
+
+    /**
+     * Force the test to use post method
+     */
+    public function post(): self
```
Comment (nunomaduro) [inline src/Factory.php]: Can you receive the payload on the post's argument instead?

File: src/Plugin.php [hunk 2]
```diff
@@ -51,12 +51,38 @@ public function handleArguments(array $arguments): array
...
+
+            if (str_starts_with($argument, '--method=')) {
+                $run->method(str_replace('--method=', '', $argument));
+            }
+
+            if ($argument === '--get') {
+                $run->get();
+            }
+
+            if ($argument === '--post') {
```
Comment (nunomaduro) [inline src/Plugin.php]: Can you receive the payload on the post's argument instead?

File: src/Plugin.php [hunk 3]
```diff
@@ -51,12 +51,38 @@ public function handleArguments(array $arguments): array
...
                 $run->duration((int) str_replace('--duration=', '', $argument));
             }
-        }
 
-        foreach ($arguments as $argument) {
             if (str_starts_with($argument, '--concurrency=')) {
                 $run->concurrently((int) str_replace('--concurrency=', '', $argument));
             }
+
+            if (str_starts_with($argument, '--method=')) {
```
Comment (nunomaduro) [inline src/Plugin.php]: Remove this.

**Review by nunomaduro — CHANGES_REQUESTED** [review]

File: src/Factory.php (hunk 1 above)
Comment (nahime0) [reply src/Factory.php]: Sure. Commit https://github.com/pestphp/pest-plugin-stressless/pull/8/commits/b7a9490b9e1aa33394f9a9ef3a396f344ddfdf9d

I've removed the also the `method` method, as it doesn't make sense now

File: src/Plugin.php (hunk 2 above)
Comment (nahime0) [reply src/Plugin.php]: Done: https://github.com/pestphp/pest-plugin-stressless/pull/8/commits/b7a9490b9e1aa33394f9a9ef3a396f344ddfdf9d

File: src/Plugin.php (hunk 3 above)
Comment (nahime0) [reply src/Plugin.php]: Done: https://github.com/pestphp/pest-plugin-stressless/pull/8/commits/b7a9490b9e1aa33394f9a9ef3a396f344ddfdf9d

**Comment by nunomaduro** [comment]
@nahime0 Are you able to PR the other verbs? Like put, delete, etc?

**Comment by nahime0** [comment]
Hi @nunomaduro, Yes, of course. I'll be engaged on a test for a job position till tomorrow noon. I'll be back on this right after.

**Comment by nahime0** [comment]
@nunomaduro Just opened a new PR https://github.com/pestphp/pest-plugin-stressless/pull/15

**Comment by SilvioDoMine** [comment]
@nunomaduro do you think is pertinent to have a option where we can stress test with a dynamic payload?

Let's say we are stress testing a payment gateway, and we want to hit the STORE payment multiple times with multiple ids. Maybe we can stress with different payload/id for each request

**Pull request merged**